src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt940Utils.java

```diff
@@ -45,139 +49,93 @@
-     * @param reference   Reference
-     * @return  Whether the reference is valid or not
-     */
-    public static boolean isValidateReference(String reference) {
-        return (StringUtils.isNotBlank(reference) && reference.length() < 17);
-    }
-
-    /**
-     * Method to validate the Sequence Number in the request payload.
-     * @param sequenceNumber   Sequence Number
-     * @return  Whether the sequence number is valid or not
-     */
-    public static boolean isValidateSequenceNumber(String sequenceNumber) {
-        return (StringUtils.isNotBlank(sequenceNumber) && sequenceNumber.length() < 4);
-    }
-
     /**
      * Method to validate the Balance Details in the request payload.
      *
      * @param balanceDetails   Balance Details
      * @param fieldName        Balance Field Name
-     * @return  Return ErrorModel if the balance is invalid.
+     * @return  Return validationResult if the balance is invalid.
      */
-    public static ErrorModel validateBalanceDetails(BalanceModel balanceDetails, String fieldName) {
-
-        if (!isValidDateFormat(balanceDetails.getDate())) {
-            return new ErrorModel(ConnectorConstants.ERROR_T50,
-                    String.format(ConnectorConstants.ERROR_INCORRECT_FORMAT, fieldName));
-        }
+    public static ValidationResult validateBalanceDetails(JSONObject balanceDetails, String fieldName) {
+
+        return JsonToMT940PayloadValidator.getMT940BalanceValidationEngine(fieldName)
+                .addCustomRule(new MT940IndicatorValidationRule(new ValidatorContext(MT940Constants.BAL_INDICATOR,
+                        ConnectorUtils.concatFieldsWithSpaces(fieldName, MT940Constants.DN_INDICATOR))))
+                .addCustomRule(new MT940StatementTypeValidationRule(new ValidatorContext(
+                        MT940Constants.BAL_STATEMENT_TYPE, ConnectorUtils.concatFieldsWithSpaces(fieldName,
+                        MT940Constants.DN_STATEMENT_TYPE))))
+                .addCustomRule(new MT940AmountFormatValidationRule(new ValidatorContext(MT940Constants.BAL_AMOUNT,
+                        ConnectorUtils.concatFieldsWithSpaces(fieldName, MT940Constants.DN_DATE))))
+                .validate(balanceDetails);
+    }
 
-        if (!ValidatorUtils.isValidCurrency(balanceDetails.getCurrency())) {
-            return new ErrorModel(ConnectorConstants.ERROR_T52,
-                    String.format(ConnectorConstants.ERROR_CURRENCY_CODE_INVALID, fieldName));
+    /**
+     * Method to validate the Balance object details.
+     *
+     * @param requestPayload  Request Payload
+     * @return              Validation Result if there is an error, else empty Validation Result
+     */
+    public static ValidationResult validateBalances(JSONObject requestPayload) {
+        ValidationResult validationResult = JsonToMt940Utils.validateBalanceDetails(requestPayload
+                        .getJSONObject(MT940Constants.OPENING_BAL_DETAILS),
```

Comment:
Hope we have done the relevant null checks before reaching this point.

Reply:
Yes all the null checks are done before